### PR TITLE
Throw error on ipfs cluster pinning when creating market

### LINKED ISSUE #213: Creating market should error if adding to IPFS cluster fails
Right now creating market just logs error if adding to cluster fails.

### PR DESCRIPTION
**Pull request by yornaath**
Fixes #213

### CODE REVIEW

File: packages/sdk/src/models/index.ts [hunk 1]
```diff
@@ -222,15 +224,28 @@ export default class Models {
         const unsub = await tx.signAndSend(
           signer.address,
           { signer: signer.signer },
-          (result) =>
+          (result) => {
+            setTimeout(() => {
+              if (result.dispatchError || result.internalError) {
+                this.ipfsClient.unpinCidFromCluster(cid.toString());
+              }
+            });
```
Comment (yornaath) [inline packages/sdk/src/models/index.ts:232]: using set timeout to prevent the call from block the callback that has higher priority.

Comment (saboonikhil) [reply packages/sdk/src/models/index.ts:232]: Should we log the result of the unpinning? Would be beneficial in catching errors if any

File: packages/sdk/src/models/index.ts [hunk 2]
```diff
@@ -143,6 +143,8 @@ export default class Models {
       })
     );
 
+    this.ipfsClient;
```
Comment (saboonikhil) [inline packages/sdk/src/models/index.ts:146]: Looks like this object is stranded?!

**Comment by saboonikhil** [comment]
Is frontend using [createMarket](https://github.com/zeitgeistpm/tools/blob/2807efc773b98785f3b6fdb46d8e23d92953a87f/packages/sdk/src/models/index.ts#L251) ? A similar change might be required for it as well.

**Review by saboonikhil — COMMENTED** [review]
Is frontend using [createMarket](https://github.com/zeitgeistpm/tools/blob/2807efc773b98785f3b6fdb46d8e23d92953a87f/packages/sdk/src/models/index.ts#L251) ? A similar change might be required for it as well.

**Review by saboonikhil — APPROVED** [review]

**Pull request merged**